### ISSUE TITLE
Datepicker on day selection resets the view to the current month

### ISSUE THREAD

**Issue by rbansal27** [gpbl/react-day-picker#2405]
[Link to Reproduce](https://codesandbox.io/p/sandbox/condescending-glitter-vk7xc4?file=%2Fsrc%2FDatepicker.tsx%3A174%2C28)

## Problem

When the date is selected for the month other than the current, Datepicker automatically resets the View to the current month. 
[Accessibility guidelines](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/) to refer.

## Expected Behavior

Datepicker should by default show the selected date irrespective of the month being current or past or future. 

## Actual Behavior
Datepicker on selection of the date is resetting the view to the current month. 

Version 9.0.8

## Screenshots

![datepicker](https://github.com/user-attachments/assets/07cdb026-3bff-4380-a218-8ebf9d3110ee)

**Comment by eminx** [comment]
I can confirm that, as a very new user, I'm experiencing the exact same problem.
Great tool by the way!

**Comment by NikosgGrigoratos** [comment]
I found this prop:
![image](https://github.com/user-attachments/assets/21154c7d-f402-4aeb-a13d-e9c17cd9fd38)

It should solve your issue. Here is your example with the prop:
https://codesandbox.io/p/sandbox/youthful-tesla-rlgddy?file=%2Fsrc%2FDatepicker.tsx%3A81%2C6

**Comment by gpbl** [comment]
Thanks here for reporting this issue, I'm working on it. A temporary workaround is to [control the month](https://daypicker.dev/docs/navigation#onmonthchange):

```tsx
export function Controlled() {
  const [month, setMonth] = useState(nextMonth);
  return ( 
     <DayPicker month={month} onMonthChange={setMonth} />
  );
}
```

**Comment by eminx** [comment]
> Thanks here for reporting this issue, I'm working on it. A temporary workaround is to [control the month](https://daypicker.dev/docs/navigation#onmonthchange):
> 
> ```tsx
> export function Controlled() {
>   const [month, setMonth] = useState(nextMonth);
>   return ( 
>      <DayPicker month={month} onMonthChange={setMonth} />
>   );
> }
> ```

Unfortunately doesn't work for me...

**Comment by wlachs** [comment]
> Thanks here for reporting this issue, I'm working on it. A temporary workaround is to [control the month](https://daypicker.dev/docs/navigation#onmonthchange):
> 
> ```tsx
> export function Controlled() {
>   const [month, setMonth] = useState(nextMonth);
>   return ( 
>      <DayPicker month={month} onMonthChange={setMonth} />
>   );
> }
> ```

This workaround only works with `numberOfMonths` < 2. Otherwise, the selected month always "jumps" to the left.

**Comment by gpbl** [comment]
This should be fixed in `v9.0.9`. Thanks for your help!

**Comment by caleb-reyes-op** [comment]
solved for me in 9.0.9, thanks!! :)
